### PR TITLE
Partially revert "srcpkgs/*: remove all cryptocurrency/blockchain packages"

### PR DESCRIPTION
**Pull request by lemmi**
I kind of hate how #44422 went down. No attempt was made to find some sort of middle ground and I wasn't able to find any sort of discussion about it. Maybe we can make up for it now.

I reverted #44422 and split it in single commits per package. If someone were to step up and take over maintenance of a package, I can revert that removal easily. But I will only do that if someone is willing to maintain a package and someone from @void-linux/pkg-committers signals support.

I'll take the first step and say I'm willing to oversee `bitcoin` and maybe `electrum`, if @ar-jan  and @damadmai (or someone else) are interested. (Which is why I left these packages in and did not add commits for them). I could probably be convinced to also oversee `go-ethereum`, since that template seems to be straight forward for now.

There are a couple of packages that were abandoned from upstream or I wasn't able to find any pull requests for, so I think they can go. I also think that we don't need to ship any miners, since at this point, it's mostly special purpose software and is probably best built directly from upstream. I also wouldn't be opposed to add the last bit to our package requirements.

Here's list of people that I was able to find that last touched some of the removed packages, so they know what's going on, or can leave comment:
 
namecoin: @Chocimier @mobinmob 
litecoin: @ahesford @ndowens 
monero: @kevcrumb @Hoshpak 
monero-gui: @kevcrumb @Hoshpak 
go-ethereum: @sihoang
bitcoin: @damadmai @unspecd  #44133
Electron-Cash: @ar-jan  #43702
electrum: @ar-jan  #43702

If this is left without any meaningful feedback for about a month, I'll close this as is. And since this is potentially a highly polarizing topic, I feel I have to ask everyone to keep it civilized in advance.

[ci-skip]

### CODE REVIEW

**Comment by damadmai** [comment]
I would be interested to keep the `bitcoin` package up to date and favor the suggestion of @ar-jan

> How about not accepting new package requests and removing unmaintained packages, but keeping actively maintained ones?

Actually I was installing and using the package for an university course to learn about cryptocurrencies:

https://tiss.tuwien.ac.at/course/courseDetails.xhtml?dswid=4235&dsrid=410&courseNr=192065&semester=2022W&locale=en

It was nice to be just able to install it with XBPS, as some other distributions didn't include it in their repositories.

**Comment by ar-jan** [comment]
I also really dislike how this was done. I actually understand that committers would prefer "no cryptocurrency" to not have to deal with this, but for existing packages that can't be the only consideration. There are users of these packages. The ones that were maintained could've been grandfathered in.

I hope we can get the Void Linux team to reconsider. If so, I'm willing to adopt `electrum`, I'd already been keeping that updated. I'd say not to keep `Electron-Cash`, since it requires a different libsecp256k library for full functionality, so it's better to run as AppImage.

**Comment by Clos3y** [comment]
I think this discussion would be better placed under the '[Discussions](https://github.com/void-linux/void-packages/discussions)' tab rather than a PR.

**Comment by kevcrumb** [comment]
I can continue to maintain `monero`. `monero-gui` is not of personal interest to me, but as long as its maintenance doesn't cause any headaches, I can adopt that one too.

Not that it is related to Void, but there's a reality of people struggling to make ends meet for whom Monero is a way to lower costs of living. I personally know a family of immigrants for whom the cost of their bank account eats a significant percentage of their income - even though it's mostly empty - but they cannot get electricity for cash, so they rely on it for payments and are now looking into Monero as alternative.

**Comment by lemmi** [comment]
So here is what I like to do: 

1. I'll revert the changes to `CONTRIBUTING.md`. They are too broad for my taste and I think we can come up with something more reasonable. 
2. Since we have updates available for `bitcoin` and `electrum`, I don't think I need to touch `removed-packages` as the removals are versioned. People can just reopen their PRs or make a new one and I'll merge them. Just ping me when you are ready for review.
3. We have no feedback for `go-ethereum`, so I'll take this as a sign, that there isn't enough interest to keep it around.
4. Even though there is interest in `monero`, I won't have the capacity to oversee that, so @kevcrumb may need to find someone else from the @void-linux/pkg-committers team to sign off on that one.
5. Later, I'll close this PR as is, in case someone wants to cherry pick some of the commits from here.

**Comment by ar-jan** [comment]
Sounds good. I've pushed a [branch for a new PR](https://github.com/void-linux/void-packages/compare/master...ar-jan:void-packages:electrum-libsecp256k1). Should I wait for a new phrasing around crypto in `CONTRIBUTING.md` before opening a PR to avoid further issues?

**Comment by lemmi** [comment]
@damadmai Still interested in maintaining `bitcoin`?

@ar-jan I'll just prepare everything locally and will push it all in one go.

**Comment by damadmai** [comment]
Yes!

**Comment by lemmi** [comment]
@damadmai cool, then please confirm that the `bitcoin` PR is up-to date (rebase, check whether there is newer version maybe, reopen or make a new one) and ping me when ready.

**Comment by kevcrumb** [comment]
> 4\. need to find someone else from the @void-linux/pkg-committers team

Is there a working format for such a ping or a compiled list of names?

Worst case, I could publish to [Split's repos](https://gitlab.com/splitlinux/split-packages/) until interest emerges ...

**Comment by damadmai** [comment]
@lemmi Yes, #44133 PR is up-to date

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -20,7 +20,6 @@ while not meeting any of the following requirements, is a good candidate for the
 In particular, new themes are highly unlikely to be accepted.
 Simple shell scripts are unlikely to be accepted unless they provide considerable value to a broad user base.
 New fonts may be accepted if they provide value beyond aesthetics (e.g. they contain glyphs for a script missing in already packaged fonts).
-Packages related to cryptocurrencies (wallets, miners, nodes, etc) are not accepted.
```
Comment (0x5c) [inline CONTRIBUTING.md:23]: No need to remove that to "grandfather-in" the packages you're planning on reinstating; the other categories have such exceptions

**Comment by sgn** [comment]
Anychange bitcoin monero namecoin gets merged soon?

I need to keep them on the tracking list for Boost

**Comment by Yorizuka** [comment]
Just leaving a comment cheering on this PR, I was not happy to see the Monero wallet removed. I am also upset at the likely politically motivated attack on what where healthy looking packages. It's one thing to not add every coin in the world, its another to remove a popular or a well maintained package for it simply being a package related to crypto.

**Comment by wilkart** [comment]
Just for the record, I also see removing maintained and working packages (I use monero thanks to @kevcrumb) this way.

Looks like decided and done by @0x5c and @leahneukirchen

I understand reasoning of building them but I have impression they should be placed in
https://docs.voidlinux.org/xbps/repositories/restricted.html
if there is really IRC based discussion and concensus of this decision.

With all respect to Void commiters and maintainers.

**Pull request closed without merging**